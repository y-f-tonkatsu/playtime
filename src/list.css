.imageListRoot {
    display: block;
    width: 100%;
    height: 100vh;
    overflow-y: scroll;
}

.imageList {
    display: flex;
    width: 100%;
}

.imageListItem {
    display: block;
    position: relative;
    overflow-y: hidden;
}


.imageListItemOverLay {
    position: absolute;
    color: rgba(250, 250, 220, 0.9);
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    opacity: 0.05;
}

.imageListItemOverLay:hover {
    background: rgba(40, 40, 60, 0.5);
    opacity: 1;
}

.imageListTitle {
    font-family: ta-f1blockline, sans-serif;
    font-size: 1.8rem;
    padding: 1rem;
}

.imageListDescription {
    font-family: noto-sans-cjk-jp, sans-serif;
    font-size: 0.9rem;
    padding: 1rem 1.5rem 0rem 1.5rem;
}

.imageListImage {
    position:relative;
    display: block;
    width: 94%;
    margin: 6px 3% ;
}

.columns {
    width: 25%;
    overflow-y: visible;
}

.imageColumn0 {
    background: #FFBF10;
}

.imageColumn1 {
    background: #FFFF00;
}

.imageColumn2 {
    background: #B7FF00;
}

.imageColumn3 {
    background: #33B600;
}
