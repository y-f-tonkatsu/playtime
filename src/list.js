const React = require('react');
const ReactDom = require('react-dom');
const Link = require('react-router-dom').Link;
const style = require('./list.css');

const path = require('path')
const _ = require('lodash');

const Header = require('./header');

const Util = require('./pt-util');

class ListItem extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            'work': props.work
        }

    }

    render() {
        const work = this.state.work;
        return (
            <Link to={'/:' + work.id}
                  className={style.imageListItem}>
                <img className={style.imageListImage}
                     src={Util.getThumbPath(work)}
                />
                <div className={style.imageListItemOverLay}>
                    <h3 className={style.imageListTitle}>{work.title}</h3>
                    <p className={style.imageListDescription}>{work.description}</p>
                </div>
            </Link>
        );
    }
}

class Column extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            works: props.works,
            columnId: props.columnId
        }
    }

    render() {

        const items = [];
        this.state.works.forEach((work) => {
            items.push(<ListItem work={work} key={work.id}/>)
        });

        return (
            <article className={[style['imageColumn' + this.state.columnId], style.columns].join(' ')}>
                {items}
            </article>
        )

    }
}

class List extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            'works': props.works
        }

    }

    getColumns() {
        const items = [[], [], [], []];
        const heights = [0, 0, 0, 0];
        let currentColumn = 0;
        this.state.works.forEach((work) => {
            currentColumn = 0;
            let tempH = heights[0];
            for (let i = 1; i < 4; i++) {
                if (tempH > heights[i]) {
                    currentColumn = i;
                    tempH = heights[i];
                }
            }

            items[currentColumn].push(work);
            heights[currentColumn] += parseInt(work.thumbHeight);
        });

        const columns = [];
        _.times(4, (i) => {
            columns.push(<Column works={items[i]} columnId={i} key={i}/>);
        });
        return columns;
    }

    render() {

        const columns = this.getColumns();

        return (
            <div className={style.imageListRoot}>
                <Header/>
                <div className={style.imageList}>
                    {columns[0]}
                    {columns[1]}
                    {columns[2]}
                    {columns[3]}
                </div>
            </div>
        );
    }
}

module.exports = {
    'List': List,
    'ListItem': ListItem,
};