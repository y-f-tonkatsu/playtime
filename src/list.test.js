import React from 'react';
import renderer from 'react-test-renderer';

import {List, ListItem} from './list';

const works = [
    {
        "id": 4,
        "title": "六本木の夜景",
        "description": "「神様はお亡くなりになった。」とマハル師はかたったが、実際は、現実的に考えればまず死んでいると見て間違いない状況であったというだけで、はっきりと死んだと分かる描写はなく、続編制作決定となれば実は生きていたことにできる余地は残っているのである。",
        "path": "/absphoto/002_roppongi.jpg",
        "lastBuildDate": "Wed, 11 Apr 2018 21:40:00 +0900"
    },
    {
        "id": 3,
        "title": "丸の内の夜景2",
        "description": "生ゆば君がニンジンを凝視して「あー、参考にならん、このモデル全然参考にならんわー。所詮ニンジンだしなー。」とかいいながら、トヨタのSUVのデッサンをしてて、ニンジンにしたら「俺に何を期待してるんだ..」って感じだろーなーと思った。",
        "path": "/absphoto/001_maru2.jpg",
        "lastBuildDate": "Wed, 11 Apr 2018 21:40:00 +0900"
    },
    {
        "id": 2,
        "title": "丸の内の夜景",
        "description": "「子は怪力乱神を語らず」と言われた孔子は、残業は敬しつつもこれを遠ざけるのが知恵であると語っている。「敬遠」の語源であり、また、孔子の合理主義者としての一面を示すエピソードである。",
        "path": "/absphoto/000_maru.jpg",
        "lastBuildDate": "Wed, 11 Apr 2018 21:40:00 +0900"
    },
    {
        "id": 1,
        "title": "歌舞伎町と果物",
        "description": "「私は天皇陛下の人格を非常にそんけいしている。」とマハル師はかたった。「なので、私は自分が天皇ではなくてほんとうによかったとおもっている。」ともかたった。生ゆば君は立派な人格など持たされては負けなのかもしれないと思った。",
        "path": "/absphoto/010_abs004.jpg",
        "lastBuildDate": "Wed, 11 Apr 2018 21:40:00 +0900"
    }
];

const work = works[works.length - 1];

test('Image List Item', () => {
    const component = renderer.create(
        <ListItem work={work} key={work.id}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});

test('Image List', () => {
    const component = renderer.create(
        <List works={works} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});