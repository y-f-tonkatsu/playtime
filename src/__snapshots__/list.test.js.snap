// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image List 1`] = `
<ul
  className="imageList"
>
  <li>
    <img
      src="works\\\\absphoto\\\\002_roppongi.jpg"
    />
  </li>
  <li>
    <img
      src="works\\\\absphoto\\\\001_maru2.jpg"
    />
  </li>
  <li>
    <img
      src="works\\\\absphoto\\\\000_maru.jpg"
    />
  </li>
  <li>
    <img
      src="works\\\\absphoto\\\\010_abs004.jpg"
    />
  </li>
</ul>
`;

exports[`Image List Item 1`] = `
<li>
  <img
    src="works\\\\absphoto\\\\010_abs004.jpg"
  />
</li>
`;
